scripts/quests/flyers_for_regine.lua

```diff
@@ -0,0 +1,89 @@
+-----------------------------------
+-- Flyers for Regine
+-----------------------------------
+require("scripts/globals/zone")
+require("scripts/globals/quests")
+require("scripts/globals/npc_util")
+-----------------------------------
+
+quests = quests or {}
+quests.flyers_for_regine = quests.flyers_for_regine or {}
+
+-------------------------------------------------
+-- local data
+-------------------------------------------------
+
+-- [ffr]deliveryMask stores delivered fliers as bits (0 = not delivered, 1 = delivered)
+-- the key in this table relates to the bit in the mask
+-- region is the region in the zone around the NPC, for trigging the approach message
+-- offset is the message that plays when approaching the NPC (offset from FFR_LOOKS_CURIOUSLY_BASE)
+
+local npcData =
+{
+    [ 0] = {zone = tpz.zone.PORT_SAN_DORIA,     region = 1, x =   76.331, z = -128.655, offset = 0, npc = 'Answald'},
+    [ 1] = {zone = tpz.zone.PORT_SAN_DORIA,     region = 2, x =  -13.768, z =  -95.857, offset = 1, npc = 'Prietta'},
+    [ 2] = {zone = tpz.zone.PORT_SAN_DORIA,     region = 3, x =    0.750, z =  -81.438, offset = 2, npc = 'Miene'},
+    [ 3] = {zone = tpz.zone.PORT_SAN_DORIA,     region = 4, x =  -21.999, z = -106.877, offset = 3, npc = 'Portaure'},
+    [ 4] = {zone = tpz.zone.PORT_SAN_DORIA,     region = 5, x =  -37.821, z =   40.949, offset = 4, npc = 'Auvare'},
+
+    [ 5] = {zone = tpz.zone.NORTHERN_SAN_DORIA, region = 2, x =  146.420, z =  127.601, offset = 0, npc = 'Coullene'},
+    [ 6] = {zone = tpz.zone.NORTHERN_SAN_DORIA, region = 3, x = -159.082, z =  253.794, offset = 1, npc = 'Guilberdrier'},
+    [ 7] = {zone = tpz.zone.NORTHERN_SAN_DORIA, region = 4, x = -156.242, z =   49.184, offset = 2, npc = 'Boncort'},
+    [ 8] = {zone = tpz.zone.NORTHERN_SAN_DORIA, region = 5, x = -127.355, z =  130.461, offset = 3, npc = 'Capiria'},
+    [ 9] = {zone = tpz.zone.NORTHERN_SAN_DORIA, region = 6, x = -157.524, z =  263.818, offset = 4, npc = 'Villion'},
+
+    [10] = {zone = tpz.zone.SOUTHERN_SAN_DORIA, region = 2, x =   33.033, z =  -30.119, offset = 0, npc = 'Blendare'},
+    [11] = {zone = tpz.zone.SOUTHERN_SAN_DORIA, region = 3, x =   69.895, z =   41.073, offset = 1, npc = 'Rosel'},
+    [12] = {zone = tpz.zone.SOUTHERN_SAN_DORIA, region = 4, x =  105.147, z =  -16.735, offset = 3, npc = 'Maugie'},
+    [13] = {zone = tpz.zone.SOUTHERN_SAN_DORIA, region = 5, x =   80.378, z =  -24.644, offset = 5, npc = 'Adaunel'},
+    [14] = {zone = tpz.zone.SOUTHERN_SAN_DORIA, region = 6, x = -136.322, z =   21.958, offset = 7, npc = 'Leuveret'},
+}
+
+-------------------------------------------------
+-- public functions
+-------------------------------------------------
+
+quests.flyers_for_regine.initZone = function(zone)
+    local zoneId = zone:getID()
+
+    for k, v in pairs(npcData) do
```

Comment:
It might be possible to reduce the number of loops by taking advantage of how every zone always has 5. You'd need a _tiny_ bridge table linking a zone ID to an offset, but once you have it:
```lua
-- pseudo
local zone_offsets = {
  [tpz.zone.PORT_SAN_DORIA] = 0,
  [tpz.zone.NORTHERN_SAN_DORIA] = 1,
}

quests.flyers_for_regine.initZone = function(zone)
local offset = ffr.zoneOffset[zone_id]
for i = offset, offset + 5 do
  local npc = npcData[i]
  registerRegion(npc.region, npc.x)
```

Although it'd only save _just a few_ cycles, so it's up to you.

Reply:
I'd like to leave this PR as is for now.  However, I'm going to revisit FFR once I have answers to questions in Captures discord.  When I do the next (last?) FFR PR, I'll include this change too.

Reply:
Sounds good - I guess I jumped the gun when coming up with a target branch name for this. 😅

Reply:
who knew Regine was a one winged angel? ¯\\_(ツ)\_\/¯